### PR TITLE
Task 6 redux

### PR DESCRIPTION
**Pull request by dhaneeshkmrt**

### CODE REVIEW

File: src/invision/components/movie-popup/movie.popup.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default function MoviePopup(props) {
   }
```
Comment (shuspes) [inline src/invision/components/movie-popup/movie.popup.js:23]: Modal form is not cleared after creating or closing Add Movie form

File: src/invision/components/movie-list/movie-list.js [hunk 2]
```diff
@@ -155,4 +155,13 @@ export default function MovieList() {
     </>
   );
 
-}
\ No newline at end of file
+}
+
+function mapToProps(store) {
```
Comment (shuspes) [inline src/invision/components/movie-list/movie-list.js:160]: You do not need this function. You do not use **fullMovieList**.

File: src/invision/components/movie-list/movie-list.js [hunk 3]
```diff
@@ -155,4 +155,13 @@ export default function MovieList() {
...
+  const { fullMovieList } = store.movies;
+  return {
+    fullMovieList
+  }
+}
+
+export default connect(mapToProps)(MovieList);
```
Comment (shuspes) [inline src/invision/components/movie-list/movie-list.js:167]: When you use Redux hooks, you do not need **connect** HOC to get access to the dispatch function. You should use **useDispatch** hook - [documentation](https://react-redux.js.org/api/hooks#usedispatch), [article](https://levelup.gitconnected.com/react-redux-hooks-useselector-and-usedispatch-f7d8c7f75cdd).

File: src/invision/components/movie-list/movie-list.js [hunk 4]
```diff
@@ -11,47 +12,57 @@ import './movie-list.scss';
...
-    updateMovieList(searchedMovies);
+    props.dispatch({ type: FILTER_CATEGORY_BY_SEARCH_KEYWORD, payload: searchKeyword })
+    // const searchedMovies = fullMovieList.filter(movie => movie.title.toLowerCase().includes(searchKeyword.toLowerCase()));
+    // updateMovieList(searchedMovies);
   }
 
   const filterCategory = (categoryName = '') => {
-    const searchedMovies = fullMovieList.filter(movie => movie.genre.toLowerCase().includes(categoryName.toLowerCase()));
-    updateMovieList(searchedMovies);
+    props.dispatch({ type: FILTER_CATEGORY_BY_NAME, payload: categoryName })
```
Comment (shuspes) [inline src/invision/components/movie-list/movie-list.js:53]: You should create actions for sort and filter functionality as you did for movies

File: src/invision/components/movie-list/movie-list.js [hunk 5]
```diff
@@ -11,47 +12,57 @@ import './movie-list.scss';
...
 
-  const [movieList, updateMovieList] = useState([]);
-  const [fullMovieList, updateFullMovieList] = useState([]);
-  const [sortValue, updateSortValue] = useState('RELEASE DATE');
-  const [isEditMovieVisible, updateMoviePopupVisibility] = useState(false);
-  const [isDeleteMovieVisible, updateDeleteMoviePopupVisibility] = useState(false);
-  const [selectedMovie, updateSelectedMovie] = useState(null);
-  const [detailedMovie, updateDetailedMovie] = useState(null);
+  useEffect(() => {
+    props.dispatch(getFullMovieList());
```
Comment (shuspes) [inline src/invision/components/movie-list/movie-list.js:24]: Please, use **useDispatch** Redux hook

File: src/invision/components/movie-list/movie-list.js [hunk 6]
```diff
@@ -11,47 +12,57 @@ import './movie-list.scss';
...
-    new MovieService().getMovieList().then(newMovieList => {
-      updateMovieList(newMovieList);
-      updateFullMovieList(newMovieList);
-    });
+  const movieList = useSelector(state => {
+    return state.movies.movieList
   });
 
+  const genres = useSelector(state => state.movies.genres);
+  // const filterBySearchKeyword = useSelector(state => state.movie.searchKeyword);
```
Comment (shuspes) [inline src/invision/components/movie-list/movie-list.js:32]: Please, delete all not needed comments and code from the PR

File: src/invision/store/reducers/movie.reducer.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      const genres = []
+      fullMovieList.forEach(movie => {
+        if (!genres.includes(movie.genre)) {
+          genres.push(movie.genre);
+        }
+      });
+      return { ...state, fullMovieList, movieList: sort([...fullMovieList], 'releasedDate'), genres: [...state.genres, ...genres] };
+
+    case ADD_MOVIE:
+      return { ...state, fullMovieList: [...state.fullMovieList, action.payload], movieList: [...state.movieList, action.payload] }
```
Comment (shuspes) [inline src/invision/store/reducers/movie.reducer.js:31]: You do not upgrate **genres** list by the value from new movie

File: src/invision/store/reducers/movie.reducer.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+    case EDIT_MOVIE: {
+      const fullMovieList = [...state.fullMovieList];
+      const index = fullMovieList.findIndex(movie => movie.id === action.payload.id);
+      fullMovieList[index] = action.payload;
+      const movieList = [...state.movieList];
+      const index1 = state.movieList.findIndex(movie => movie.id === action.payload.id);
+      movieList[index1] = action.payload;
+      return { ...state, fullMovieList, movieList };
```
Comment (shuspes) [inline src/invision/store/reducers/movie.reducer.js:40]: You do not sort movies again if needed value was changed in the movie

File: src/invision/store/reducers/popup.reducer.js [hunk 9]
```diff
@@ -0,0 +1,10 @@
+import { ADD_MOVIE_POPUP } from "../actions/action.types";
+export function popupReducer(state = {}, action) {
```
Comment (shuspes) [inline src/invision/store/reducers/popup.reducer.js:2]: This reducer is not used in the app

File: src/invision/store/actions/add-movie.action.js [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import { ADD_MOVIE } from "./action.types";
```
Comment (shuspes) [inline src/invision/store/actions/add-movie.action.js:1]: Usually, all actions for one reducer are collected to the one file. If you want to split action functions to the separate files, put them in one folder that may be easily connected with reducer

File: src/invision/store/actions/add-movie.action.js [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+import { movieMapper } from '../../utils/movie-mapper.util';
+import { movieConverter } from '../../utils/movie-converter.util';
+
+export const addMovieAction = (movie) => {
+  movie = movieConverter(movie);
+  return dispatch => {
+    fetch(`http://localhost:4000/movies?sortBy=release_date&sortOrder=asc`, {
```
Comment (shuspes) [inline src/invision/store/actions/add-movie.action.js:8]: Do you really need this URL in the **addMovieAction** action?

File: src/invision/utils/movie-converter.util.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
+export const movieConverter= (movie)=>{
```
Comment (shuspes) [inline src/invision/utils/movie-converter.util.js:1]: I like your **movieMapper** and **movieConverter**

**Review by shuspes — APPROVED** [review]

File: src/invision/store/actions/add-movie.action.js (hunk 11 above)
Comment (dhaneeshkmrt) [reply src/invision/store/actions/add-movie.action.js:8]: this URL is provided in the API documentation. So I used that.

File: src/invision/utils/movie-converter.util.js (hunk 12 above)
Comment (dhaneeshkmrt) [reply src/invision/utils/movie-converter.util.js:1]: thanks

File: src/invision/store/actions/add-movie.action.js (hunk 10 above)
Comment (dhaneeshkmrt) [reply src/invision/store/actions/add-movie.action.js:1]: Sure. I will follow this hereafter.

**Pull request merged**